### PR TITLE
Dala 2463 implement cookie banner

### PR DESCRIPTION
**Pull request by marcelrasbachdfine**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] The version of main currently active on prod is deployed to a dev server with `Reset non-user related Docker Volumes & Re-populate` turned on
  - [ ] It's verified that the CD run is green
  - [ ] The data from prod is migrated using `./migrateData.sh dataland.com <SOURCE_API_KEY> <TARGET> <TARGET_API_KEY>` 
  - [ ] The feature branch is deployed to the same server with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the dev server
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server
- [x] Make sure that pull requests in the development tools repository, which are connected to this pull requests, 
are merged

### CODE REVIEW

File: dataland-frontend/.env [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #Dataland settings to source into frontend code
 VITE_DATA_REQUEST_UPLOAD_MAX_FILE_SIZE_IN_MEGABYTES=${DATA_REQUEST_UPLOAD_MAX_FILE_SIZE_IN_MEGABYTES}
 VITE_DOCUMENT_UPLOAD_MAX_FILE_SIZE_IN_MEGABYTES=${DOCUMENT_UPLOAD_MAX_FILE_SIZE_IN_MEGABYTES}
-VITE_MAX_NUMBER_OF_DAYS_SELECTABLE_FOR_API_KEY_VALIDITY=${MAX_NUMBER_OF_DAYS_SELECTABLE_FOR_API_KEY_VALIDITY}
+VITE_MAX_NUMBER_OF_DAYS_SELECTABLE_FOR_API_KEY_VALIDITY=${MAX_NUMBER_OF_DAYS_SELECTABLE_FOR_API_KEY_VALIDITY}
```
Comment (marcelrasbachdfine) [inline dataland-frontend/.env:4]: there should not be any change to this file in the final version

Comment (stewal19) [reply dataland-frontend/.env:4]: Why is this flagged as changed anyway? Seems rather identical

Comment (marcelrasbachdfine) [reply dataland-frontend/.env:4]: i added and removed a line in some previous commits, but have on idea actually

File: dataland-frontend/tests/e2e/specs/framework-view-and-upload-journey/EditDataset.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ describeIf(
       cy.fixture("CompanyInformationWithLksgPreparedFixtures").then(function (jsonContent) {
         const preparedFixtures = jsonContent as Array<FixtureData<LksgData>>;
         preparedFixture = getPreparedFixture("lksg-all-fields", preparedFixtures);
+        cy.pause();
```
Comment (ahoecherl) [inline dataland-frontend/tests/e2e/specs/framework-view-and-upload-journey/EditDataset.ts:27]: Please add a comment that indicates why this pause is added?

File: dataland-frontend/tests/e2e/specs/landing-page/CookieBanner.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        cy.visitAndCheckAppMount("/");
+        cy.get("div[id='CybotCookiebotDialog']").should("exist");
+        cy.get("#CybotCookiebotDialogPoweredbyImage").invoke("attr", "src").should("contain", "data:image/png;base64,");
+        cy.get("button[id='CybotCookiebotDialogBodyLevelButtonLevelOptinAllowAll']").should("exist");
+        cy.get("button[id='CybotCookiebotDialogBodyLevelButtonLevelOptinAllowallSelection']").should("exist");
+        cy.get("button[id='CybotCookiebotDialogBodyButtonDecline']").should("exist");
+        cy.get("div[id='CybotCookiebotDialogBodyFieldsetInnerContainer']").should("exist");
+        cy.get("h2[id='CybotCookiebotDialogBodyContentTitle']")
+          .should("exist")
+          .should("contain", "This website uses cookies");
```
Comment (ahoecherl) [inline dataland-frontend/tests/e2e/specs/landing-page/CookieBanner.ts:20]: Might not work anymore after changing the cookiebanner to German.

File: dataland-frontend/src/components/resources/newLandingPage/TheQuotes.vue [hunk 4]
```diff
@@ -18,7 +18,16 @@
...
+            class="cookieconsent-optout-marketing"
+            :src="'https://www.youtube-nocookie.com/embed/' + card.icon + '?rel=0'"
+            title="Youtube video player"
+            allowfullscreen
+            :class="{ 'quotes__slide-video--zoom-out': currentSlide !== index - 1, 'quotes__slide-video': true }"
+          ></iframe>
+          <iframe
+            class="cookieconsent-optin-marketing"
+            :data-src="'https://www.youtube.com/embed/' + card.icon + '?rel=0'"
+            data-cookieconsent="marketing"
```
Comment (ahoecherl) [inline dataland-frontend/src/components/resources/newLandingPage/TheQuotes.vue:30]: Does this work like a show-if? I'm surprised that both the no-cookie and the cookie iframe appear to be part of the page at all times.

File: dataland-frontend/tests/e2e/specs/framework-view-and-upload-journey/EditDataset.ts (hunk 2 above)
Comment (marcelrasbachdfine) [reply dataland-frontend/tests/e2e/specs/framework-view-and-upload-journey/EditDataset.ts:27]: yes, that's how it works. 
the css class is created by coookiebot.com
(If i wanted to do this with v-if i would need to ask for consent anyways and contact cookiebot for that)

File: dataland-frontend/src/components/resources/newLandingPage/TheQuotes.vue (hunk 4 above)
Comment (marcelrasbachdfine) [reply dataland-frontend/src/components/resources/newLandingPage/TheQuotes.vue:30]: yes, that's how it works.
the css class is created by coookiebot.com
(If i wanted to do this with v-if i would need to ask for consent anyways and contact cookiebot for that)

File: dataland-frontend/tests/e2e/specs/landing-page/CookieBanner.ts (hunk 3 above)
Comment (marcelrasbachdfine) [reply dataland-frontend/tests/e2e/specs/landing-page/CookieBanner.ts:20]: i will change this to German

Comment (marcelrasbachdfine) [reply dataland-frontend/tests/e2e/specs/landing-page/CookieBanner.ts]: pushed

File: dataland-frontend/tests/e2e/specs/framework-view-and-upload-journey/EditDataset.ts (hunk 2 above)
Comment (marcelrasbachdfine) [reply dataland-frontend/tests/e2e/specs/framework-view-and-upload-journey/EditDataset.ts]: also pushed

**Pull request merged**